app/src/main/java/com/example/ncov19traking/data/NCoVRepository.kt

```diff
@@ -1,86 +1,117 @@
+import com.example.ncov19traking.api.*
+import com.example.ncov19traking.models.*
 import java.io.IOException
 
-class NCoVRepository(private val context: Context) {
+class NCoVRepository(nCoVDataBase: NCoVDataBase) {
 
-    private val nCoVDao = NCoVDataBase.getDataBase(context).nCoVDao()
-    private val countryDao = NCoVDataBase.getDataBase(context).countryDao()
-    private val globalHistoricalDao = NCoVDataBase.getDataBase(context).globalHistoricalDao()
+    private val nCoVDao = nCoVDataBase.nCoVDao()
+    private val countryDao = nCoVDataBase.countryDao()
+    private val globalHistoricalDao = nCoVDataBase.globalHistoricalDao()
+    private var errorResponse = MutableLiveData<ErrorBody>()
 
     suspend fun getAllCases(): NCoVInfo {
         return try {
-            refreshAllCases()
-            nCoVDao.load().first()
+            when (val response = ApiResponse.create(NCoVApiClient.nCoVApi.getGeneralNumbers())) {
+                is ApiSuccessResponse -> {
+                    nCoVDao.save(response.data)
+                    nCoVDao.load().first()
+                }
+                is ApiSuccessEmptyResponse -> nCoVDao.load().first()
+                is ApiErrorResponse -> {
+                    Log.d("apiError", "${response.code}: ${response.message}")
+                    errorResponse.postValue(ErrorBody(response.code, response.message))
+                    nCoVDao.load().first()
+                }
+            }
         } catch (e: IOException) {
-            if (nCoVDao.load().size != 0)
+            errorResponse.postValue(e.message?.let { ErrorBody(message = it) })
+            if (nCoVDao.load().isNotEmpty())
                 nCoVDao.load().first()
             else NCoVInfo(0, 0, 0, 0)
         }
     }
 
-    private suspend fun refreshAllCases() {
-            NCoVApiAdapter.nCoVApi.getGeneralNumbers().body()?.let { nCoVDao.save(it) }
-    }
-
     fun deleteAllCases() {
         nCoVDao.delete()
     }
 
     suspend fun getAllYesterdayCases(): NCoVInfoYesterday {
         return try {
-            refreshAllYesterdayCases()
-            nCoVDao.loadYesterday().first()
+            when (val response =
+                ApiResponse.create(NCoVApiClient.nCoVApi.getYesterdayGeneralNumbers())) {
+                is ApiSuccessResponse -> {
+                    nCoVDao.saveYesterday(response.data)
+                    nCoVDao.loadYesterday().first()
+                }
+                is ApiSuccessEmptyResponse -> nCoVDao.loadYesterday().first()
+                is ApiErrorResponse -> {
+                    Log.d("apiError", "${response.code}: ${response.message}")
+                    nCoVDao.loadYesterday().first()
+                }
+            }
         } catch (e: Exception) {
-            if (nCoVDao.loadYesterday().size != 0)
+            if (nCoVDao.loadYesterday().isNotEmpty())
                 nCoVDao.loadYesterday().first()
             else
                 NCoVInfoYesterday(0, 0, 0, 0)
```

Comment:
why not set default values to the `NCoVInfoYesterday` constructor parameters so you can just initialize it with an empty constructor.

Reply:
fixed